File: JS/ProjectEuler10.js

var isPrime = function (number) {
    for (var i = 2; i <= Math.sqrt(number); i++) {
        if (number % i === 0) {
            return false;
        }
    }
    return true;
}

var getList = function (to,func) {
    var result = [];
    for (var i = 2; i < to; i++) {
        if (func(i)) {
            result.push(i);
        }
    }
    return result;
}

console.log(getList(2000000,isPrime).reduce(function(x, y) {
    return x + y;
}));

while (true) { }

File: FSharp/ProjectEuler10.fs
let isPrime number = [2L..( int64(sqrt (double number)))] |> List.filter(fun x -> number % x = 0L) |> List.length = 0

let primeSeq = Seq.unfold(fun(x)-> Some(x, x+1L))(2L) |> Seq.takeWhile(fun x -> x < 2000000L) |> Seq.filter(isPrime) 

[<EntryPoint>]
let main argv =
    let result = primeSeq |> Seq.sum
    printfn "%A" result
    0 // return an integer exit code
